horepgd.py

```diff
@@ -69,22 +71,28 @@ def run_import(wanted_channels, tvhsocket, fetch_radio=False, nr_days=5):
         listings = Listings()
         # add listings for each of the channels
         for channel_id, channel in chmap.channel_map.items():
-            if channel['title'].lower() in (channel.lower() for channel in wanted_channels):
+            title = channel['title']
+            if title.lower() in (channel.lower() for channel in wanted_channels) or title.lower().replace(' hd', '') in (channel.lower() for channel in wanted_channels):
                 now = datetime.date.today().timetuple()
                 number = 0
                 xmltv = XMLTVDocument()
                 # channel logo
                 icon = None
                 for asset in channel['images']:
                     if asset['assetType'] == 'station-logo-large':
-                        icon = asset['url']
+                        u = urlparse(asset['url'])
+                        icon = u.scheme + "://" + u.netloc + u.path
+                        debug('Found icon {:s}'.format(icon))
                         break
+                if title.lower().replace(' hd', '') != title.lower() and title.lower().replace(' hd', '') in (channel.lower() for channel in wanted_channels):
+                    title = title.replace(' HD', '')
```

Comment:
This will not replace correctly when `HD` is not capitalized in the channel title. Also, consider where the phrase ` hd` is part of another section of the title that does not denote HDTV.

Reply:
I've replaced this by checking if the channel endswith(" hd") the title is set with a substring [:-3]